pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,451 @@
+			want:     true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := dtc.IsDeviceExist(test.deviceID); got != test.want {
+				t.Errorf("DTContext.IsDeviceExist() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+//Function TestGetDevice is to test GetDevice().
+func TestGetDevice(t *testing.T) {
+	dtc := &DTContext{
+		DeviceList: &sync.Map{},
+	}
+	var device dttype.Device
+	dtc.DeviceList.Store("FailureCaseDeviceID", "FailureCaseCasevalue")
+	dtc.DeviceList.Store("SuccessCaseDeviceID", &device)
+	tests := []struct {
+		name     string
+		deviceID string
+		want     *dttype.Device
+		wantbool bool
+	}{
+		{
+			//DeviceID not present
+			name:     "FailureCaseNoDeviceID",
+			deviceID: "FailureCaseID",
+			want:     nil,
+			wantbool: false,
+		},
+		{
+			//DeviceID present but unable to get device
+			name:     "FailCaseTestNoDevice",
+			deviceID: "FailureCaseDeviceID",
+			want:     nil,
+			wantbool: false,
+		},
+		{
+			name:     "SuccessCaseTest",
+			deviceID: "SuccessCaseDeviceID",
+			want:     &dttype.Device{},
+			wantbool: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, gotbool := dtc.GetDevice(test.deviceID)
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("DTContext.GetDevice() got = %v, want %v", got, test.want)
+			}
+			if gotbool != test.wantbool {
+				t.Errorf("DTContext.GetDevice() gotbool = %v, want %v", gotbool, test.wantbool)
+			}
+		})
+	}
+}
+
+//Function TestSend is function to test Send().
+func TestSend(t *testing.T) {
+	commChan := make(map[string]chan interface{})
+	testinterface := make(chan interface{}, 1)
+	commChan[dtcommon.TwinModule] = testinterface
+
+	var msg *model.Message
+	dtc := &DTContext{
+		CommChan: commChan,
+	}
+
+	tests := []struct {
+		name     string
+		identity string
+		action   string
+		module   string
+		msg      *model.Message
+		wantErr  bool
+	}{
+		{
+			name:     "FailureCaseTest",
+			identity: "",
+			action:   dtcommon.SendToCloud,
+			module:   dtcommon.CommModule,
+			msg:      msg,
+			wantErr:  true,
+		},
+		{
+			name:     "SuccessCaseTest",
+			identity: "",
+			action:   dtcommon.SendToCloud,
+			module:   dtcommon.TwinModule,
+			msg:      msg,
+			wantErr:  false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if err := dtc.Send(test.identity, test.action, test.module, test.msg); (err != nil) != test.wantErr {
+				t.Errorf("DTContext.Send() error = %v, wantErr %v", err, test.wantErr)
+			}
+		})
+	}
+}
+
+//TestBuildModelMessage is to test BuildModelMessage().
+func TestBuildModelMessage(t *testing.T) {
+	dtc := &DTContext{}
+	tests := []struct {
+		name      string
+		group     string
+		parentID  string
+		resource  string
+		operation string
+		content   interface{}
+		want      *model.Message
+	}{
+		{
+			name:      "BuildModelMessageTest",
+			group:     "resource",
+			parentID:  "",
```

Comment:
default value will be "", no need to fill i guess ? Applicable to all cases below